core/chain_indexer.go

```diff
@@ -376,8 +380,94 @@ func (c *ChainIndexer) PruneOldBlockData(blockHeight uint64) {
 	rawdb.DeletePendingEtxsRollup(c.chainDb, blockHash)
 	rawdb.DeleteManifest(c.chainDb, blockHash)
 	rawdb.DeletePbCacheBody(c.chainDb, blockHash)
-	rawdb.DeleteSpentUTXOs(c.chainDb, blockHash)
+	createdUtxos, _ := rawdb.ReadCreatedUTXOKeys(c.chainDb, blockHash)
+	createdUtxosToKeep := make([][]byte, 0, len(createdUtxos))
+	for _, key := range createdUtxos {
+		data, _ := c.chainDb.Get(key)
+		if len(data) == 0 {
+			// Don't keep it if it doesn't exist
+			continue
+		}
+		utxoProto := new(types.ProtoTxOut)
+		if err := proto.Unmarshal(data, utxoProto); err != nil {
```

Comment:
again, how is this scenario possible, except in the event of a very bad bug?

The consequences here are scary. If we change the proto encoder without properly handling the legacy format (e.g. upgrading the legacy format to the new format) for every stored UTXO, this code will proceed to delete all UTXOs. One easy bug from us results in permanent money deletion for everyone... this seems scary.

Reply:
You are looking at an old commit. This is no longer the case.

Reply:
Also, you are misunderstanding the function - this function prunes old key data, removing to-be-trimmed keys from the set if they aren't eligible to be trimmed (namely because the denomination is too high)